src/zmqsubscriber/zmqsubscriber.go

```diff
@@ -109,14 +121,27 @@ func (z *ZMQSubscriber) Run() error {
 
 		topic, payload := string(msg[0]), msg[1:]
 		log.Printf(`[ZMQ] received topic "%s"`, topic)
-		switch topic {
-		case TOPIC_HASHTX:
-			z.IncomingTx <- parseTransaction(t, payload)
-		case TOPIC_RAWBLOCK:
-			z.IncomingBlocks <- parseBlock(t, payload)
-		default:
-			return fmt.Errorf("unknown topic %s", topic)
-		}
+
+
+		// process received messages asynchronously so that we do not stall the queue while parsing
+		go func() {
```

Comment:
Could be a `handleZMQMessage(topic, payload)` function. No need to have it as a anonymous inline function I think.

Reply:
to clarify, I think this would be more readable and still testable

```go
go func(topic, payload) {
  if err := z.processMessage(topic, payload); err != nil {
    parseErrors <- err
  }
}(topic, payload)
```

passing `chan error` to other funcs complicates their implementation